atlasdb-commons/src/test/java/com/palantir/common/base/ThrowablesTest.java

```diff
@@ -61,7 +71,46 @@ public void testRewrap() {
             int sizeAfter = e.getStackTrace().length;
             assertTrue(sizeAfter + " should be > " + sizeBefore, sizeAfter > sizeBefore);
         }
+    }
+
+    @Test
+    public void unwrapAndThrowRateLimitExceededOrAtlasDbDependencyExceptionCanThrowRateLimitExceededException() {
+        assertThatThrownBy(() -> Throwables.unwrapAndThrowRateLimitExceededOrAtlasDbDependencyException(
+                RATE_LIMIT_EXCEEDED_EXCEPTION)).isEqualTo(RATE_LIMIT_EXCEEDED_EXCEPTION);
+    }
+
+    @Test
+    public void unwrapAndThrowRateLimitExceededOrAtlasDbDependencyExceptionCanThrowAtlasDbDependencyException() {
+        assertThatThrownBy(() -> Throwables.unwrapAndThrowRateLimitExceededOrAtlasDbDependencyException(
+                ATLASDB_DEPENDENCY_EXCEPTION)).isEqualTo(ATLASDB_DEPENDENCY_EXCEPTION);
+    }
 
+    @Test
+    public void unwrapAndThrowRateLimitExceededOrAtlasDbDependencyExceptionThrowsWrappedRateLimitExceededExceptions() {
+        assertThatThrownBy(() -> Throwables.unwrapAndThrowRateLimitExceededOrAtlasDbDependencyException(
+                new ExecutionException(RATE_LIMIT_EXCEEDED_EXCEPTION))).isEqualTo(RATE_LIMIT_EXCEEDED_EXCEPTION);
+        assertThatThrownBy(() -> Throwables.unwrapAndThrowRateLimitExceededOrAtlasDbDependencyException(
+                new InvocationTargetException(RATE_LIMIT_EXCEEDED_EXCEPTION))).isEqualTo(RATE_LIMIT_EXCEEDED_EXCEPTION);
+    }
+
+    @Test
+    public void unwrapAndThrowRateLimitExceededOrAtlasDbDependencyExceptionThrowsWrappedAtlasDbDependencyExceptions() {
```

Comment:
Great job on adding tests for these, some were beyond the scope of this PR :)

This one test looks repeated, can we delete this.

Reply:
Actually this test should have used the `ATLASDB_DEPENDENCY_EXCEPTION`, I went too fast and forgot to change it :) good spot!